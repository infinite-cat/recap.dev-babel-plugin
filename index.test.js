const pluginTester = require('babel-plugin-tester').default;
const yourPlugin = require('./index');

pluginTester({
  plugin: yourPlugin,
  pluginName: 'identifier reverse',
  tests: [
    {
      code: `
        function sayHi(person) {
          // do some work
          5 + 4;
          const a = 145;
          a + 4;
          return 'Hello ' + person + '!'
        }

        function noReturn() {
          console.log('test')
        }

        const testArrow = () => {
          const x = 3 + 5
          console.log(x)
          return x
        }

        class TestClass {
          testMethod() {
            console.log('test class', this)
          }
        }

        sayHi()
        noReturn()

      `,
      snapshot: true,
    },
  ],
})
