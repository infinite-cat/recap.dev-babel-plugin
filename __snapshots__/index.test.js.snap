// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`identifier reverse 1. identifier reverse: 1. identifier reverse 1`] = `

function sayHi(person) {
  // do some work
  5 + 4;
  const a = 145;
  a + 4;
  return 'Hello ' + person + '!'
}

function noReturn() {
  console.log('test')
}

const testArrow = () => {
  const x = 3 + 5
  console.log(x)
  return x
}

class TestClass {
  testMethod() {
    console.log('test class', this)
  }
}

sayHi()
noReturn()

      ↓ ↓ ↓ ↓ ↓ ↓

if (typeof autotracer !== "function") {
  var autotracer = require("performance-measurer/autotracer");
}

function sayHi(person) {
  // do some work
  5 + 4;
  const a = 145;
  a + 4;
  return "Hello " + person + "!";
}

sayHi = autotracer.wrapFunction("unknown file", "sayHi", sayHi);

function noReturn() {
  console.log("test");
}

noReturn = autotracer.wrapFunction("unknown file", "noReturn", noReturn);
const testArrow = autotracer.wrapFunction("unknown file", "testArrow", () => {
  const x = 3 + 5;
  console.log(x);
  return x;
});

class TestClass {
  testMethod() {
    console.log("test class", this);
  }
}

autotracer.wrapClass("unknown file", "TestClass", TestClass);
sayHi();
noReturn();


`;
